kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -106,7 +108,9 @@ class KyuubiSessionImpl(
           Option(password).filter(_.nonEmpty).getOrElse("anonymous")
         }
       try {
-        _client = KyuubiSyncThriftClient.createClient(user, passwd, host, port, sessionConf)
+        val engineSessionConf = sessionConf.clone
+        engineSessionConf.set(KyuubiReservedKeys.KYUUBI_ENGINE_CREDENTIALS_KEY, engineCredentials)
```

Comment:
The `engineSessionConf` doesn't seem to be passed to engine, put it in `optimizedConf`?

Reply:
Seems we can not put it into optimizedConf, it might be passed to SparkProcessBuilder and will be passed to engine.

Reply:
How about:
```
        sessionConf.set(KyuubiReservedKeys.KYUUBI_ENGINE_CREDENTIALS_KEY, engineCredentials)
        _client = KyuubiSyncThriftClient.createClient(user, passwd, host, port, sessionConf)
```
cc @yaooqinn

Reply:
got it, thanks

Reply:
Sorry, I didn't express my idea exactly. I mean that we need to pass this configuration through the 'openSession' method instead of the 'createClient'`.